spec/jobs/reports/mfa_report_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe Reports::MfaReport do
+  let(:issuer) { 'issuer1' }
+  let(:issuers) { [issuer] }
+  let(:report_date) { Date.new(2023, 12, 0o1) }
+  let(:email)  { 'partner.name@example.com' }
+  let(:name) { 'Partner Name' }
+
+  let(:report_configs) do
+    [
+      {
+        'name' => name,
+        'issuers' => issuers,
+        'emails' => [email],
+      },
+    ]
+  end
+
+  before do
+    allow(IdentityConfig.store).to receive(:s3_reports_enabled).and_return(true)
+    allow(IdentityConfig.store).to receive(:mfa_report_config) { report_configs }
+  end
+
+  describe '#perform' do
+    let(:reports) { double('emailable_reports').as_null_object }
+
+    let(:monthly_report) { double(Reporting::MfaReport, as_emailable_reports: reports) }
+
+    before do
+      expect(Reporting::MfaReport).to receive(:new).with(
+        issuers:,
+        time_range: report_date.all_month,
+      ) { monthly_report }
```

Comment:
I feel like we learned the hard way that you should have one test that is an integration test which does not stub out the class in `Reporting` and at most uses the `stub_cloudwatch_logs`. Copying the `stub_cloudwatch_logs` call that already exists in spec/lib/reporting/mfa_report_spec.rb should be fine

Reply:
Do we have an example of this in one of the other specs?